tests/sentry/issues/test_occurrence_consumer.py

```diff
@@ -83,3 +83,41 @@ def test_invalid_occurrence_payload(self) -> None:
         message = get_test_message(self.project.id, type=300)
         occurrence = _process_message(message)
         assert occurrence is None
+
+
+class ParseEventPayloadTest(IssueOccurrenceTestMessage):
+    def run_test(self, message):
+        _process_message(message)
+
+    def run_invalid_schema_test(self, message):
+        with pytest.raises(InvalidEventPayloadError):
+            self.run_test(message)
+
+    def run_invalid_payload_test(self, remove_event_fields=None, update_event_fields=None):
+        message = deepcopy(get_test_message(self.project.id))
+        if remove_event_fields:
+            for field in remove_event_fields:
+                message["event"].pop(field)
+        if update_event_fields:
+            message["event"].update(update_event_fields)
+        self.run_invalid_schema_test(message)
+
+    def test_invalid_payload(self):
+        # self.run_invalid_payload_test(remove_event_fields=["event_id"])
+        # self.run_invalid_payload_test(remove_event_fields=["project_id"])
+        # self.run_invalid_payload_test(remove_event_fields=["timestamp"])
+        # self.run_invalid_payload_test(remove_event_fields=["platform"])
+        # self.run_invalid_payload_test(remove_event_fields=["tags"])
```

Comment:
these don't work right now because we're manually parsing the message payload before validating the schema and sending the message to snuba. not sure if we want to mix up that order or if i'm missing something here